dingo/core/utils/plotting.py

```diff
@@ -0,0 +1,108 @@
+from itertools import zip_longest
+
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import corner
+import numpy as np
+import pandas as pd
+
+
+def plot_corner_multi(
+    samples, weights=None, labels=None, filename="corner.pdf", **kwargs
+):
+    """
+    Generate a corner plot for multiple posteriors.
+
+    Parameters
+    ----------
+    samples : list[pd.DataFrame]
+        List of sample sets. The DataFrame column names are used as parameter labels.
+    weights : list[np.ndarray or None] or None
+        List of weights sets. The length of each array should be the same as the length of
+        the corresponding samples.
+    labels : list[str or None] or None
+        Labels for the posteriors.
+    filename : str
+        Where to save samples.
+    **kwargs :
+        Forwarded to corner.corner.
+    """
+    # Define plot properties
+    cmap = "Dark2"
+    corner_params = {
+        "smooth": 1.0,
+        "smooth1d": 1.0,
+        "plot_datapoints": False,
+        "plot_density": False,
+        "plot_contours": True,
+        "levels": [0.5, 0.9],
+        "bins": 30,
+    }
+    corner_params.update(kwargs)
+
+    serif_old = mpl.rcParams["font.family"]
+    mpl.rcParams["font.family"] = "serif"
+    linewidth_old = mpl.rcParams["lines.linewidth"]
+    mpl.rcParams["lines.linewidth"] = 2.5
+
+    # In case a single corner plot is desired, convert to lists to iterate.
+    if not isinstance(samples, list):
+        samples = [samples]
+    if not isinstance(weights, list):
+        weights = [weights]
+    if not isinstance(labels, list):
+        labels = [labels]
+
+    fig = None
+    handles = []
+    for i, (s, w, l) in enumerate(zip_longest(samples, weights, labels)):
+        color = mpl.colors.rgb2hex(plt.get_cmap(cmap)(i))
+        fig = corner.corner(
+            s.to_numpy(),
+            labels=s.columns,
+            weights=w,
+            color=color,
+            no_fill_contours=True,
```

Comment:
Maybe we want to allow for filled contours here, should this go in corner_params? Or maybe this only goes to hist_2d? When I set it to False nothing happens anyway.

Reply:
I experimented with filled contours, and I did not like how they looked unfortunately. I tried to make them semi-transparent as we currently have them, using `contourf_kwargs = {"alpha": 0.3}`, but this did not work. (There seems to be a bug, see https://github.com/dfm/corner.py/issues/112).

If you can get it working, please feel free to do so.

Reply:
Also, to turn on the filling (aside from setting alpha) you need to remove the `no_fill_contours` argument, and set `fill_contours=True`.

Reply:
Ah, I see there is that bug.

For now maybe I can just set the default to false in ```corner_params``` and then the user can update it w/ ```corner_params.update(kwargs)```. How does it sound?

Reply:
Yeah, the user can update the kwargs as it stands now. But for filling the contours we may want to do this selectively for each sample set (especially with alpha = 1).

I also left one of the old ChainConsumer plotting functions, which now imports the package in a try/except block. So it's still there. In fact, one could set Dingo to use ChainConsumer in all cases when it is available. But that is a bit more work.

Reply:
Sounds good!